File: src/main/java/com/github/matteo099/model/dao/DieDao.java
package com.github.matteo099.model.dao;

import java.util.List;

import com.github.matteo099.model.entities.DieType;
import com.github.matteo099.model.entities.MaterialType;
import com.github.matteo099.model.interfaces.ICustomer;
import com.github.matteo099.model.interfaces.IDie;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DieDao implements IDie<DieShapeExport, ICustomer> {
    public String name;
    public DieShapeExport dieData;
    public String customer;
    public List<String> aliases;
    public DieType dieType;
    public MaterialType material;
    public Double totalHeight;
    public Double totalWidth;
    public Double shoeWidth;
    public Double crestWidth;

    @Override
    public ICustomer getCustomer() {
        return new ICustomer() {
            @Override
            public String getName() {
                return customer;
            }
        };
    }
}

File: src/main/webui/src/app/models/dao/die-dao.ts
import { DieType } from "../entities/die.type";
import { MaterialType } from "../entities/material-type";
import { DieDataDao } from "./die-data-dao";

export interface DieDao {
    name: string;
    dieData: DieDataDao;
    aliases: string[];
    customer: string;

    dieType: DieType;
    material: MaterialType;
    totalHeight: number;
    totalWidth: number;
    shoeWidth: number;
    crestWidth: number;
}